File: Portal.Services/AzW.Model/Provision/NSGRule.cs

using System.Collections.Generic;

namespace AzW.Model
 {
     public class NSGRule: Resource
     {
          public string Direction { get; set; } //"in", "out"
          public bool Allow { get; set; }
          public string FromAddresses { get; set; }
          public string FromPorts { get; set; }
          public string ToAddresses { get; set; }
          public string ToPorts { get; set; }
          public string Protocol { get; set; } //tcp, udp, icmp, *
          public int Priority { get; set; }
     }
 }

File: src/Portal/src/models/NSGRule.js
export default class NSGRule
{
    constructor()
    {
       this.UITableRowID =  '';
       this.Name = '';
       this.Allow = false;
       this.FromAddressIsTag = false;
       this.FromAddresses = '';
       this.FromPorts = '*';
       this.ToAddressIsTag = false;
       this.ToAddresses = '';
       this.ToPorts = '*';
       this.Protocol = '*'; //tcp, udp, icmp, *
       this.Priority = 100; //100 and 4096
    }
}